mailbox/memory/src/test/java/org/apache/james/mailbox/inmemory/manager/InMemoryIntegrationResources.java

```diff
@@ -172,6 +173,13 @@ default FinalStage quotaManager(QuotaManager quotaManager) {
         }
 
         interface FinalStage {
+            @FunctionalInterface
+            interface InstanciateListener {
```

Comment:
What it this type's intent?

Reply:
This interface allow to instantiate a MailboxListener from main mailbox components